t/APISIX.pm

```diff
@@ -777,7 +779,7 @@ _EOC_
             set \$upstream_connection         '';
 
             set \$upstream_scheme             'http';
-            set \$upstream_host               \$http_host;
+            set \$upstream_host               \$host;
```

Comment:
The value of variable $http_host header will be empty string when using http/3
ref: https://github.com/zulip/zulip/pull/26348